text/0258-env-context-baggage-carriers.md

```diff
@@ -0,0 +1,294 @@
+for this form of propagation.
+
+Notably, as we define semantic conventions within the [CI/CD Working Group][cicd-wg],
+we'll need the specification defined for the industry to be able to adopt
+native tracing wtihin CI/CD systems.
+
+[cicd-wg]: https://github.com/open-telemetry/community/blob/main/projects/ci-cd.md
+[issue-740]: https://github.com/open-telemetry/opentelemetry-specification/issues/740#issue-665588273
+
+## Design
+
+To propagate context and baggage between parent, sibling, and child processes
+in systems where HTTP communication does not occur between processes, a
+specification using key-vaulue pairs injected into the environment can be read
+and produced by an arbitrary TextMapPropagator.
+
+### Example Context
+
+Consider the following diagram in the context of process forking:
+
+> Note: The diagram simply an example and simplification of process forking.
+> There are other ways to spawn processes which are more performant like
+> exec().
+
+<!-- TODO: Maybe change diagram to not show fork. -->
+
+![Environment Variable Context Propagation](./img/0258-env-context-parent-child-process.png)
+
+In the above diagram, a parent process is forked to spawn a child process,
+inheriting the environment variables from the original parent. The environment
+variables defined here, `TRACEPARENT`, `TRACESTATE`, and `BAGGAGE` are used to
+propagate context to the child process such that it can be tied to the parent.
+Without `TRACEPARENT`, a tracing backend would not be able to connect the child
+process spans to the parent span, forming an end-to-end trace.
+
+> Note: While the below exclusively references the W3C Specification, this
+> proposal is not exclusive to W3C and is instead focused on the mechanism of
+> Text Map Propagation with a potential set of well-known environment variable
+> names.
+
+`traceparent` (lowercase), originates in the [W3C Specification][w3c-parent]
+and includes the following fields:
+
+* `version`
+* `trace-id`
+* `parent-id`
+* `trace-flags`
+
+This could be set in the environment as follows:
+
+```bash
+export TRACEPARENT=version=2HEXDIGLC,trace-id=32HEXDIGLC,parent-id=16HEXDIGLC,trace-flags=2HEXDIGLC
+```
+
+`tracestate` (lowercase), originates in the [W3C Specification][w2c-state] and
+can include any opaque value in a key-value pair structure. Its goal is to
+provide additional vendor-specific trace information.
+
+`baggage` (lowercase), also is defined in the [W3C Specification][w3c-bag] and
+is a set of key-value pairs to propagate context between signals. In
+OpenTelemetry, baggage is propagated through the [Baggage API][bag-api].
+
+[w3c-parent]: https://www.w3.org/TR/trace-context-2/#traceparent-header-field-values
+[w3c-state]: https://www.w3.org/TR/trace-context-2/#tracestate-header
+[w3c-bag]: https://www.w3.org/TR/baggage/#baggage-http-header-format
+
+#### Distributed Tracing in OpenTofu Prototype Example
+
+Consider this real world example OpenTofu Controller Deployment.
```

Comment:
Maybe add a few words to the end of this section, that there is an ongoing discussion in the OpenTofu project around introducing OpenTelemetry support (thanks @apparentlymart for linking this OTEP and the OpenTofu RFC [here](https://github.com/opentofu/opentofu/pull/2028#issuecomment-2397631674))